lib/mix/lib/mix/scm/git.ex

```diff
@@ -52,11 +52,12 @@ defmodule Mix.SCM.Git do
         File.cd!(opts[:dest], fn ->
           rev_info = get_rev_info
```

Comment:
@josevalim I implemented this (calling git only once) in aa5276f; I think the implementation looks a little bit more noisy, but I agree this may be worth it.